cou/utils/app_utils.py

```diff
@@ -25,17 +26,24 @@
 logger = logging.getLogger(__name__)
 
 
-async def upgrade_packages(units: Iterable[str], model: COUModel) -> None:
+async def upgrade_packages(
+    units: Iterable[str], model: COUModel, packages_to_hold: Optional[list] = None
```

Comment:
maybe it's safer if we remove the default value to `None` so we force to always pass `packages_to_hold`?

Reply:
As discussed elsewhere, since we always need to specify `packages_to_hold` when using this function in Application classes for inheritance purposes, I removed its default value here.